File: Puppet/Puppet/MSPushViewController.m
/*
 * Copyright (c) Microsoft Corporation. All rights reserved.
 */

#import "MSPushViewController.h"
#import "AppCenterPush.h"

@interface MSPushViewController ()

@property (weak, nonatomic) IBOutlet UISwitch *enabled;

@end

@implementation MSPushViewController

- (void)viewDidLoad {
  [super viewDidLoad];
  self.enabled.on = [MSPush isEnabled];
}

- (IBAction)enabledSwitchUpdated:(UISwitch *)sender {
  [MSPush setEnabled:sender.on];
  sender.on = [MSPush isEnabled];
}

@end

File: TestApps/Sasquatch/Sasquatch/ViewControllers/MSPushViewController.swift
import UIKit

class MSPushViewController: UITableViewController, AppCenterProtocol {

  @IBOutlet weak var enabled: UISwitch!
  var appCenter: AppCenterDelegate!

  override func viewDidLoad() {
    super.viewDidLoad()
    self.enabled.isOn = appCenter.isPushEnabled()
  }

  @IBAction func enabledSwitchUpdated(_ sender: UISwitch) {
    appCenter.setPushEnabled(sender.isOn)
    sender.isOn = appCenter.isPushEnabled()
  }
}
